src/PrestaShopBundle/Command/GenerateMailTemplatesCommand.php

```diff
@@ -0,0 +1,112 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Command;
+
+use PrestaShop\PrestaShop\Core\Exception\InvalidException;
+use PrestaShopBundle\Service\Mail\MailTemplateGenerator;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Configuration;
+use Context;
+use Currency;
+use Language;
+use Validate;
+
+class GenerateMailTemplatesCommand extends ContainerAwareCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setName('prestashop:mail:generate')
+            ->setDescription('Generate mail templates for a specified theme')
+            ->addArgument('theme', InputArgument::REQUIRED, 'Theme to use for mail templates.')
+            ->addArgument('locale', InputArgument::REQUIRED, 'Which locale to use for the templates.')
+            ->addArgument('outputFolder', InputArgument::REQUIRED, 'Output folder to export templates.')
+        ;
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     *
+     * @return int|void|null
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $theme = $input->getArgument('theme');
+        $outputFolder = $input->getArgument('outputFolder');
+        if (!file_exists($outputFolder)) {
+            $outputFolder = implode(DIRECTORY_SEPARATOR, [getcwd(), $outputFolder]);
+            if (file_exists($outputFolder)) {
+                $outputFolder = realpath($outputFolder);
+            }
+        }
+
+        $this->initContext();
+
+        $locale = $input->getArgument('locale');
+        $language = $this->getLanguage($locale);
+
+        $output->writeln(sprintf('Exporting mail with theme %s for language %s to %s', $theme, $language->name, $outputFolder));
+        /** @var MailTemplateGenerator $catalog */
+        $generator = $this->getContainer()->get('prestashop.service.mail.mail_template_generator');
+        $generator->generateThemeTemplates($theme, $language, $outputFolder);
+    }
+
+    /**
+     * Initialize PrestaShop Context
+     */
+    private function initContext()
+    {
+        require $this->getContainer()->get('kernel')->getRootDir() . '/../config/config.inc.php';
+    }
+
+    /**
+     * @param string $locale
+     *
+     * @return Language
+     * @throws InvalidException
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    private function getLanguage($locale)
```

Comment:
this shouldn't be located in your command, this is why you need legacy classes here :)

Reply:
I'm pretty sure you have this information in Context Adapter, or you can create your own and inject it :)

Reply:
new `LanguageInterface` integrated and replaced in the code